specification/Metadata/README.md

```diff
@@ -0,0 +1,699 @@
+<!-- omit in toc -->
+# Cesium 3D Metadata Specification
+
+<!-- omit in toc -->
+## Contributors
+
+* Peter Gagliardi, Cesium
+* Sean Lilley, Cesium
+* Sam Suhag, Cesium
+* Bao Tran, Cesium
+* Patrick Cozzi, Cesium
+
+<!-- omit in toc -->
+## Status
+
+Draft
+
+<!-- omit in toc -->
+## Contents
+
+- [Overview](#overview)
+- [Concepts](#concepts)
+- [Schemas](#schemas)
+  - [Classes and Properties](#classes-and-properties)
+  - [Property Types](#property-types)
+  - [Arrays](#arrays)
+  - [Enums](#enums)
+  - [Normalized Properties](#normalized-properties)
+  - [Minimum and Maximum Values](#minimum-and-maximum-values)
+  - [Optional Properties and Default Values](#optional-properties-and-default-values)
+  - [Semantics](#semantics)
+- [Storage Formats](#storage-formats)
+  - [Table Format](#table-format)
+    - [Binary Encoding](#binary-encoding)
+      - [Numbers](#numbers)
+      - [Booleans](#booleans)
+      - [Strings](#strings)
+      - [Enums](#enums-1)
+      - [Arrays](#arrays-1)
+    - [JSON Encoding](#json-encoding)
+      - [Numbers](#numbers-1)
+      - [Booleans](#booleans-1)
+      - [Strings](#strings-1)
+      - [Enums](#enums-2)
+      - [Arrays](#arrays-2)
+  - [Raster Format](#raster-format)
+    - [Numbers](#numbers-2)
+    - [Enums](#enums-3)
+    - [Arrays](#arrays-3)
+    - [Implementation Notes](#implementation-notes)
+- [Revision History](#revision-history)
+
+## Overview
+
+The Cesium 3D Metadata Specification defines a standard metadata format for 3D data. This spans a myriad of domains, such as heights of buildings in a city, names of different structures in a CAD model, material properties for textured surfaces, and classification codes for point clouds.
+
+This specification defines a set of core concepts to be used by multiple 3D formats. It is language and format independent. This document occasionally uses JSON examples for illustrating concepts but does not define a JSON schema.
```

Comment:
While the document "does not define a JSON schema", there is really quite a lot of schema here. 😅

If I understand correctly, the purpose of this document is to define the primitive data types and relationships that will be used in `3DTILES_metadata` and `EXT_feature_metadata`, in such a way that they can be (1) referenced in both, and (2) trivially portable and compatible between the two. Is that correct?

Assuming so, in the interest of clearer delineation between the three specifications, do you think it would be reasonable to replace all or most of the JSON schemas (in this document) with structured text equivalents? For example:

**Before:**

```json
{
  "schema": {
    "classes": {
      "car": {
        "properties": {
          "forwardDirection": {
            "description": "A forward direction vector",
            "type": "ARRAY",
            "componentType": "FLOAT64",
            "componentCount": 3
          },
          "passengers": {
            "name": "Passenger Names",
            "description": "There are a variable number of passengers because componentCount is undefined.",
            "type": "ARRAY",
            "componentType": "STRING",
          },
          "modelMatrix": {
            "description": "A 4x4 model matrix",
            "type": "ARRAY",
            "componentType": "FLOAT32",
            "componentCount": 16
          }
        }
      }
    }
  }
}
```


**After:**

#### Schema → Class → Car

| name | description | type | componentType | componentCount |
|---|---|---|---|---|
|  | "A forward direction vector" | 'ARRAY' | 'FLOAT64' | 3 |
| "Passenger Names" | "Variable number of passengers, because componentCount is undefined." | 'ARRAY' | 'STRING' | |
|  | "A 4x4 model matrix" | 'ARRAY' | 'FLOAT32' | 16 |

For me this is much easier to read, I'm hopping back and forth between the three specifications quite a bit as it is, and there are some discrepancies in the JSON schema shown now.

Reply:
Additional use case: this specification defines the raster format as a 2D grid of pixels, but you can easily imagine doing the same thing with a 3D volume – for example texture arrays or video, entities having 3D coordinates associated. Certainly not something that should be "in scope" at this stage, but which would benefit from some flexibility on the encoding later on.

Reply:
:+1: for the structured texture version

Reply:
Will open a PR!

Reply:
>If I understand correctly, the purpose of this document is to define the primitive data types and relationships that will be used in 3DTILES_metadata and EXT_feature_metadata, in such a way that they can be (1) referenced in both, and (2) trivially portable and compatible between the two. Is that correct?

Correct, the purpose of this document is to factor out the common data types and encodings used in both extensions so we're not repeating ourselves too much. As you say, portability and compatibility are goals here.

Note that a specific extension that references this document might restrict attention given the limitations of the format. For example, `3DTILES_metadata` focuses more on the JSON encodings, while `EXT_feature_metadata` focuses on the binary encodings.

>this specification defines the raster format as a 2D grid of pixels, but you can easily imagine doing the same thing with a 3D volume

Huh that's a great point. 3D volumes certainly have their applications (voxels for example), haven't thought much about video or other time-dynamic data. And you could go further, think n-dimensional NumPy arrays of data for example.

For this common document that's focused on concepts, I agree that we should try to be flexible here. Let the specs that reference this document handle restricting the number of dimensions.